merlin/spec/specification.py

```diff
@@ -230,69 +230,8 @@ def dump(self):
         """
         Dump this MerlinSpec to a pretty yaml string.
         """
-        tab = "   "
-        list_offset = "  "
-        from copy import deepcopy
-
-        def dict_to_yaml(obj, string, key_stack, newline=True):
-            if obj is None:
-                return ""
-            lvl = len(key_stack) - 1
-            if isinstance(obj, str):
-                split = obj.splitlines()
-                if len(split) > 1:
-                    obj = "|\n" + tab * (lvl + 1) + ("\n" + tab * (lvl + 1)).join(split)
-                return obj
-            if isinstance(obj, bool):
-                return str(obj).lower()
-            if (not isinstance(obj, list)) and (not isinstance(obj, dict)):
-                return obj
-            if isinstance(obj, list):
-                n = len(obj)
-                use_hyphens = key_stack[-1] in ["paths", "sources", "git", "study"]
-                if not use_hyphens:
-                    string += "["
-                else:
-                    string += "\n"
-                for i, elem in enumerate(obj):
-                    key_stack = deepcopy(key_stack)
-                    key_stack.append("elem")
-                    if use_hyphens:
-                        string += (
-                            (lvl + 1) * tab
-                            + "- "
-                            + str(dict_to_yaml(elem, "", key_stack))
-                            + "\n"
-                        )
-                    else:
-                        string += str(
-                            dict_to_yaml(elem, "", key_stack, newline=(i != 0))
-                        )
-                        if n > 1 and i != len(obj) - 1:
-                            string += ", "
-                    key_stack.pop()
-                if not use_hyphens:
-                    string += "]"
-            if isinstance(obj, dict):
-                if len(key_stack) > 0 and key_stack[-1] != "elem":
-                    string += "\n"
-                i = 0
-                for k, v in obj.items():
-                    key_stack = deepcopy(key_stack)
-                    key_stack.append(k)
-                    if len(key_stack) > 1 and key_stack[-2] == "elem" and i == 0:
-                        # string += (tab * (lvl - 1))
-                        string += ""
-                    elif "elem" in key_stack:
-                        string += list_offset + (tab * lvl)
-                    else:
-                        string += tab * (lvl + 1)
-                    string += str(k) + ": " + str(dict_to_yaml(v, "", key_stack)) + "\n"
-                    key_stack.pop()
-                    i += 1
-            return string
-
-        result = dict_to_yaml(self.yaml_sections, "", [])
+        tab = 3 * " "
```

Comment:
tab is 3 spaces? I thought it's usually 4...

Reply:
Yeah, no idea what the convention is in yaml. Whatever is prettier

Reply:
@AlexanderWinterLLNL I'm in favor of decomposition, but both versions are hard to follow at a glance. So can you verify that the "pretty" provenence spec generated by your changes is the same as before your changes?